rpc-darpc/src/main/java/com/ibm/crail/namenode/rpc/darpc/DaRPCNameNodeServer.java

```diff
@@ -2,46 +2,55 @@
 
 import java.net.InetSocketAddress;
 import java.net.URI;
-
 import org.slf4j.Logger;
-
 import com.ibm.crail.conf.CrailConfiguration;
 import com.ibm.crail.rpc.RpcNameNodeService;
 import com.ibm.crail.rpc.RpcServer;
 import com.ibm.crail.utils.CrailUtils;
+import com.ibm.darpc.DaRPCMemPool;
+import com.ibm.darpc.DaRPCMemPoolImpl;
 import com.ibm.darpc.DaRPCServerEndpoint;
 import com.ibm.darpc.DaRPCServerGroup;
 import com.ibm.disni.rdma.RdmaServerEndpoint;
 
 public class DaRPCNameNodeServer extends RpcServer {
 	private static final Logger LOG = CrailUtils.getLogger();
-	
+
 	private RpcNameNodeService service;
 	private DaRPCServerGroup<DaRPCNameNodeRequest, DaRPCNameNodeResponse> namenodeServerGroup;
-	private RdmaServerEndpoint<DaRPCServerEndpoint<DaRPCNameNodeRequest, DaRPCNameNodeResponse>> namenodeServerEp;	
-	
+	private RdmaServerEndpoint<DaRPCServerEndpoint<DaRPCNameNodeRequest, DaRPCNameNodeResponse>> namenodeServerEp;
+
 	public DaRPCNameNodeServer(RpcNameNodeService service){
 		this.service = service;
 		this.namenodeServerEp = null;
 		this.namenodeServerGroup = null;
-	}	
+	}
 
 	public void init(CrailConfiguration conf, String[] args) throws Exception{
 		DaRPCConstants.updateConstants(conf);
 		DaRPCConstants.verify();
-		
+
+		DaRPCMemPool memPool = new DaRPCMemPoolImpl(
+				DaRPCConstants.NAMENODE_DARPC_MEMPOOL_HUGEPAGEPATH,
+				DaRPCConstants.NAMENODE_DARPC_MEMPOOL_ALLOCSZ,
+				DaRPCConstants.NAMENODE_DARPC_MEMPOOL_ALIGNMENT
+				);
 		String _clusterAffinities[] = DaRPCConstants.NAMENODE_DARPC_AFFINITY.split(",");
 		long clusterAffinities[] = new long[_clusterAffinities.length];
 		for (int i = 0; i < clusterAffinities.length; i++){
 			int affinity = Integer.decode(_clusterAffinities[i]).intValue();
 			clusterAffinities[i] = 1L << affinity;
 		}
 		DaRPCServiceDispatcher darpcService = new DaRPCServiceDispatcher(service);
-		this.namenodeServerGroup = DaRPCServerGroup.createServerGroup(darpcService, clusterAffinities, -1, DaRPCConstants.NAMENODE_DARPC_MAXINLINE, DaRPCConstants.NAMENODE_DARPC_POLLING, DaRPCConstants.NAMENODE_DARPC_RECVQUEUE, DaRPCConstants.NAMENODE_DARPC_SENDQUEUE, DaRPCConstants.NAMENODE_DARPC_POLLSIZE, DaRPCConstants.NAMENODE_DARPC_CLUSTERSIZE);
+		if (!DaRPCConstants.NAMENODE_DARPC_STATS.isEmpty()) {
```

Comment:
Maybe it makes sense to treat crail.namenode.darpc.stats as a boolean or similar. I can see this being misinterpreted otherwise and set to "crail.namenode.darpc.stats false" or "crail.namenode.darpc.stats no".